feature/gribi/otg_tests/base_leader_election_test/base_leader_election_test.go

```diff
@@ -128,55 +144,88 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
-		WithDefaultGateway(dutPort2.IPv4)
-
-	p3 := ate.Port(t, "port3")
-	i3 := top.AddInterface(atePort3.Name).WithPort(p3)
-	i3.IPv4().
-		WithAddress(atePort3.IPv4CIDR()).
-		WithDefaultGateway(dutPort3.IPv4)
-
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) gosnappi.Config {
+
+	otg := ate.OTG()
+	top := otg.NewConfig(t)
+
+	top.Ports().Add().SetName(atePort1.Name)
+	dev := top.Devices().Add().SetName(atePort1.Name)
+	eth := dev.Ethernets().Add().SetName(atePort1.Name + ".Eth")
+	eth.SetPortName(dev.Name()).SetMac(atePort1.MAC)
+	ip := eth.Ipv4Addresses().Add().SetName(dev.Name() + ".IPv4")
+	ip.SetAddress(atePort1.IPv4).SetGateway(dutPort1.IPv4).SetPrefix(int32(atePort1.IPv4Len))
+
+	top.Ports().Add().SetName(atePort2.Name)
+	dev = top.Devices().Add().SetName(atePort2.Name)
+	eth = dev.Ethernets().Add().SetName(atePort2.Name + ".Eth")
+	eth.SetPortName(dev.Name()).SetMac(atePort2.MAC)
+	ip = eth.Ipv4Addresses().Add().SetName(dev.Name() + ".IPv4")
+	ip.SetAddress(atePort2.IPv4).SetGateway(dutPort2.IPv4).SetPrefix(int32(atePort2.IPv4Len))
+
+	top.Ports().Add().SetName(atePort3.Name)
+	dev = top.Devices().Add().SetName(atePort3.Name)
+	eth = dev.Ethernets().Add().SetName(atePort3.Name + ".Eth")
+	eth.SetPortName(dev.Name()).SetMac(atePort3.MAC)
+	ip = eth.Ipv4Addresses().Add().SetName(dev.Name() + ".IPv4")
+	ip.SetAddress(atePort3.IPv4).SetGateway(dutPort3.IPv4).SetPrefix(int32(atePort3.IPv4Len))
+
+	otg.PushConfig(t, top)
+	otg.StartProtocols(t)
 	return top
 }
 
+func waitOtgArpEntry(t *testing.T) {
+	ate := ondatra.ATE(t, "ate")
+	ate.OTG().Telemetry().InterfaceAny().Ipv4NeighborAny().LinkLayerAddress().Watch(
+		t, time.Minute, func(val *otgtelemetry.QualifiedString) bool {
+			return val.IsPresent()
+		}).Await(t)
+}
+
 // testTraffic generates traffic flow from source network to
 // destination network via srcEndPoint to dstEndPoint and checks for
 // packet loss.
-func testTraffic(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology, srcEndPoint, dstEndPoint *ondatra.Interface) {
-	ethHeader := ondatra.NewEthernetHeader()
-	ipv4Header := ondatra.NewIPv4Header()
-	ipv4Header.DstAddressRange().
-		WithMin("198.51.100.0").
-		WithMax("198.51.100.254").
-		WithCount(250)
-
-	flow := ate.Traffic().NewFlow("Flow").
-		WithSrcEndpoints(srcEndPoint).
-		WithDstEndpoints(dstEndPoint).
-		WithHeaders(ethHeader, ipv4Header)
-
-	ate.Traffic().Start(t, flow)
+func testTraffic(t *testing.T, ate *ondatra.ATEDevice, config gosnappi.Config, srcEndPoint, dstEndPoint attrs.Attributes) {
+
+	// srcEndPoint is atePort1
+	otg := ate.OTG()
+	gwIp := inputMap[srcEndPoint].IPv4
+	waitOtgArpEntry(t)
+	dstMac := otg.Telemetry().Interface(srcEndPoint.Name + ".Eth").Ipv4Neighbor(gwIp).LinkLayerAddress().Get(t)
+	config.Flows().Clear().Items()
+	flowipv4 := config.Flows().Add().SetName("Flow")
+	flowipv4.Metrics().SetEnable(true)
+	flowipv4.TxRx().Port().SetTxName(srcEndPoint.Name).SetRxName(dstEndPoint.Name)
+	flowipv4.Size().SetFixed(512)
```

Comment:
Can we elide any of these settings like size and rate if they have a reasonable default?

Reply:
Using default 64 for size, 1000 for pps and "continuous" for duration now